### PR TITLE
Λrrow 0.7.1

### PR DESCRIPTION
**Pull request by Guardiola31337**
- Adapts code to Λrrow `0.7.1` and add [`cel4j-artifact:0.5.0`](https://github.com/macroing/CEL4J-Artifact) Java script engine
- Bump Λnk dependencies to `0.7.1` and add Kotlin, Java ~and Android `jar` and `aar` (not working)~ doc samples

👀 @pablisco @raulraja @pakoito

### CODE REVIEW

**Comment by Guardiola31337** [comment]
> `aar` (not working) 

@raulraja do you know how can we add support for Android libraries (`aar`) and Android in general? It'd be awesome if Λnk would have full support for Kotlin, Java and Android and I'm pretty sure you have some ideas to tackle this.

**Comment by pakoito** [comment]
There's no interpreter that supports Android framework unless you pack Roboelectric or something similar, right?

**Comment by raulraja** [comment]
Everything needs to go through the script engine and then you'll need the runtime libs as part of the ScriptEngine jars as well. How does one construct a classpath that is AAR and Android Capable?

**Comment by Guardiola31337** [comment]
I removed adding Android support (`aar`) from this PR because we're changing gears on how to tackle it - after a bunch of testing without luck we're going to explore @raulraja's option 👇 in a different PR.

> Potentially we could have a separate `main` for Android that launches a headless App and runs within itself the ScriptEngine.

So this is ready for a second round of 👀 @raulraja @pablisco @pakoito

File: ank-core/build.gradle [hunk 1]
```diff
@@ -20,17 +20,19 @@ kotlin {
...
+    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:$kotlinxCoroutinesVersion"
+    implementation "org.jetbrains:markdown:0.1.25"
+    implementation group: 'com.github.born2snipe', name: 'cli-progress', version: '0.4'
+    implementation 'io.arrow-kt:arrow-free:0.7.1'
+    implementation 'io.arrow-kt:arrow-instances-core:0.7.1'
+    implementation 'io.arrow-kt:arrow-data:0.7.1'
+    implementation 'io.arrow-kt:arrow-instances-data:0.7.1'
+    implementation "org.jetbrains.kotlin:kotlin-compiler:$parent.ext.kotlinVersion"
+    implementation "org.jetbrains.kotlin:kotlin-script-util:$parent.ext.kotlinVersion"
+//  implementation "org.apache-extras.beanshell:bsh:2.0b6"
```
Comment (pablisco) [inline ank-core/build.gradle]: unused dependency?

File: ank-core/build.gradle [hunk 2]
```diff
@@ -20,17 +20,19 @@ kotlin {
...
-    compile 'io.arrow-kt:arrow-data:0.6.1'
-    compile 'io.arrow-kt:arrow-syntax:0.6.1'
-    compile "org.jetbrains.kotlin:kotlin-compiler:$parent.ext.kotlinVersion"
-    compile "org.jetbrains.kotlin:kotlin-script-util:$parent.ext.kotlinVersion"
-    compile "org.apache-extras.beanshell:bsh:2.0b6"
-    kapt    'io.arrow-kt:arrow-annotations-processor:0.6.1'
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$parent.ext.kotlinVersion"
+    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:$kotlinxCoroutinesVersion"
+    implementation "org.jetbrains:markdown:0.1.25"
+    implementation group: 'com.github.born2snipe', name: 'cli-progress', version: '0.4'
```
Comment (pablisco) [inline ank-core/build.gradle]: Is it worth making it match the other dependencies?

File: ank-core/src/main/kotlin/arrow/ank/interpreter.kt [hunk 3]
```diff
@@ -32,20 +30,20 @@ val extensionMappings = mapOf(
...
-                    is AnkOps.CreateTarget -> ME.catch({ createTargetImpl(op.source, op.target) })
-                    is AnkOps.GetFileCandidates -> ME.catch({ getFileCandidatesImpl(op.target) })
-                    is AnkOps.ReadFile -> ME.catch({ readFileImpl(op.source) })
-                    is AnkOps.ParseMarkdown -> ME.catch({ parseMarkDownImpl(op.markdown) })
-                    is AnkOps.ExtractCode -> ME.catch({ extractCodeImpl(op.source, op.tree) })
-                    is AnkOps.CompileCode -> ME.catch({ compileCodeImpl(op.snippets, op.compilerArgs) })
-                    is AnkOps.ReplaceAnkToLang -> ME.catch({ replaceAnkToLangImpl(op.compilationResults) })
-                    is AnkOps.GenerateFiles -> ME.catch({ generateFilesImpl(op.candidates, op.newContents) })
-                } as HK<F, A>
+                    is AnkOps.CreateTarget -> catch { createTargetImpl(op.source, op.target) }
```
Comment (pablisco) [inline ank-core/src/main/kotlin/arrow/ank/interpreter.kt]: should we make all the catch have this format? (no parentheses)

**Comment by Guardiola31337** [comment]
@pablisco I've addressed all your comments 😉

**Pull request merged**